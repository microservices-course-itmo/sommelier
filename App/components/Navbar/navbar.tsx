import React, { ReactNode } from 'react'
import * as Styled from './navbar.styled'

type Props = {
  children: ReactNode | ReactNode[]
}

const Navbar = ({ children }: Props) => (
  <Styled.NavbarContainer>
    <Styled.ItemsContainer>{children}</Styled.ItemsContainer>
  </Styled.NavbarContainer>
)

export default Navbar
