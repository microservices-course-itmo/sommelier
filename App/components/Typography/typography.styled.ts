import styled from 'styled-components/native'
import {
  TypographyProps,
  StyledHeaderProps,
  StyledTextProps,
} from './typography.types'

const StyledTypography = styled.Text<TypographyProps>`
  color: ${({ color, theme }) =>
    color ? theme.colors[color] : theme.colors.black};
  font-weight: ${({ bold }) => (bold ? 'bold' : 'normal')};
  font-style: ${({ italic }) => (italic ? 'italic' : 'normal')};
  text-decoration: ${({ crossedOut }) =>
    crossedOut ? 'line-through' : 'none'};
`

export const StyledHeader = styled(StyledTypography)<StyledHeaderProps>`
  font-size: ${({ size, theme }) => theme.headerSizes[size]};
`

export const StyledText = styled(StyledTypography)<StyledTextProps>`
  font-size: ${({ size, theme }) => theme.fontSizes[size]};
`
