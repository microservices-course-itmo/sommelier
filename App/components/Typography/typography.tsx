import React, { ReactNode } from 'react'
import * as Styled from './typography.styled'
import theme from '../../theme/index'
import { ReactElement } from 'react'

type Props = {
  children: ReactNode
  color: string
  bold?: boolean
  italic: boolean
  size?: keyof typeof theme.fontSizes
  crossedOut?: boolean
  fontFamily: string
}

export const H1 = ({ color, children, bold, italic, fontFamily }: Props) => (
  <Styled.StyledH1
    color={color}
    bold={bold}
    italic={italic}
    fontFamily={fontFamily}
  >
    {children}
  </Styled.StyledH1>
)

export const H2 = ({ color, children, bold, italic, fontFamily }: Props) => (
  <Styled.StyledH2
    color={color}
    bold={bold}
    italic={italic}
    fontFamily={fontFamily}
  >
    {children}
  </Styled.StyledH2>
)

export const H3 = ({ color, children, bold, italic, fontFamily }: Props) => (
  <Styled.StyledH3
    color={color}
    bold={bold}
    italic={italic}
    fontFamily={fontFamily}
  >
    {children}
  </Styled.StyledH3>
)

export const H4 = ({ color, children, bold, italic, fontFamily }: Props) => (
  <Styled.StyledH4
    color={color}
    bold={bold}
    italic={italic}
    fontFamily={fontFamily}
  >
    {children}
  </Styled.StyledH4>
)

export const H5 = ({ color, children, bold, italic, fontFamily }: Props) => (
  <Styled.StyledH5
    color={color}
    bold={bold}
    italic={italic}
    fontFamily={fontFamily}
  >
    {children}
  </Styled.StyledH5>
)

export const H6 = ({ color, children, bold, italic, fontFamily }: Props) => (
  <Styled.StyledH6
    color={color}
    bold={bold}
    italic={italic}
    fontFamily={fontFamily}
  >
    {children}
  </Styled.StyledH6>
)

export const Text = ({
  children,
  color,
  bold,
  italic,
  size,
  crossedOut,
  fontFamily,
}: Props): ReactElement => (
  <Styled.StyledText
    color={color}
    bold={bold}
    italic={italic}
    fontFamily={fontFamily}
    size={size}
    crossedOut={crossedOut}
  >
    {children}
  </Styled.StyledText>
)
