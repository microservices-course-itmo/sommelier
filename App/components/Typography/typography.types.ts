import { ReactNode } from 'react'
import { COLOR_KEYS } from '../../theme/colors'
import { HEADER_SIZES_KEYS } from '../../theme/headerSizes'
import { FONT_SIZE_KEYS } from '../../theme/fontSizes'

type TypographyProps = {
  bold?: boolean
  italic?: boolean
  crossedOut?: boolean
  color?: keyof typeof COLOR_KEYS
}

export type StyledHeaderProps = TypographyProps & {
  size: keyof typeof HEADER_SIZES_KEYS
}
export type HeaderProps = StyledHeaderProps & { children: ReactNode }

export type StyledTextProps = TypographyProps & {
  size: keyof typeof FONT_SIZE_KEYS
}
export type TextProps = StyledTextProps & { children: ReactNode }
