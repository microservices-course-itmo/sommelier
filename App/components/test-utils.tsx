import React from 'react'
import { render } from '@testing-library/react-native'
import { ThemeProvider } from 'styled-components'
import theme from '../theme'

const themeRender = (component: any) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
}

export * from '@testing-library/react-native'

export { themeRender }
