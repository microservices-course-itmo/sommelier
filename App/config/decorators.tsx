import { ThemeProvider } from 'styled-components'
import React from 'react'
import styled from 'styled-components/native'
import { theme } from '../theme'

const StyledBufferView = styled.View`
  flex: 1;
  margin-vertical: 40;
  margin-horizontal: 20;
`

export const ThemeDecorator = (storyFn: any) => (
  <ThemeProvider theme={theme}>{storyFn()}</ThemeProvider>
)

export const BufferView = (storyFn: any) => (
  <StyledBufferView>{storyFn()}</StyledBufferView>
)
