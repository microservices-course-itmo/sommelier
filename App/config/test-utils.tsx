import React, { ReactNode } from 'react'
import { render } from '@testing-library/react-native'
import { ThemeProvider } from 'styled-components/native'
import { theme } from '../theme'

const themeRender = (component: ReactNode) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
}

export * from '@testing-library/react-native'

export { themeRender }
