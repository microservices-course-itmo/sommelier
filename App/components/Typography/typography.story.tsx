import React from 'react'
import styled from 'styled-components/native'
import { storiesOf } from '@storybook/react-native'
import { boolean, select } from '@storybook/addon-knobs'
import { H1, H2, H3, H4, H5, H6, Text } from './typography'
import { BufferView, ThemeDecorator } from '../../config/decorators'
import theme from '../../theme/index'

const StyledWrapper = styled.View`
  display: flex;
  flex-direction: column;
`

const FONT_SIZES = Object.keys(theme.fontSizes) as Array<
  keyof typeof theme.fontSizes
>
const COLORS = Object.keys(theme.colors) as Array<keyof typeof theme.colors>
const FONTS = Object.keys(theme.fonts) as Array<keyof typeof theme.fonts>

export const list = () => (
  <StyledWrapper>
    <H1 color='black' bold={false} italic={false} fontFamily='PtSans'>
      H1 value
    </H1>
    <H2 color='black' bold={false} italic={false} fontFamily='PtSans'>
      H2 value
    </H2>
    <H3 color='black' bold={false} italic={false} fontFamily='PtSans'>
      H3 value
    </H3>
    <H4 color='black' bold={false} italic={false} fontFamily='PtSans'>
      H4 value
    </H4>
    <H5 color='black' bold={false} italic={false} fontFamily='PtSans'>
      H5 value
    </H5>
    <H6 color='black' bold={false} italic={false} fontFamily='PtSans'>
      H6 value
    </H6>
    <Text
      color='black'
      bold={false}
      italic={false}
      fontFamily='PtSans'
      size='md'
    >
      Text value
    </Text>
  </StyledWrapper>
)

export const knobs = () => (
  <StyledWrapper>
    <H1
      color={select('color', COLORS, COLORS[1])}
      bold={boolean('bold', false)}
      italic={boolean('font-style: italic', false)}
      fontFamily={select('font-family', FONTS, FONTS[0])}
    >
      H1 value
    </H1>
    <H2
      color={select('color', COLORS, COLORS[1])}
      bold={boolean('bold', false)}
      italic={boolean('font-style: italic', false)}
      fontFamily={select('font-family', FONTS, FONTS[0])}
    >
      H2 value
    </H2>
    <H3
      color={select('color', COLORS, COLORS[1])}
      bold={boolean('bold', false)}
      italic={boolean('font-style: italic', false)}
      fontFamily={select('font-family', FONTS, FONTS[0])}
    >
      H3 value
    </H3>
    <H4
      color={select('color', COLORS, COLORS[1])}
      bold={boolean('bold', false)}
      italic={boolean('font-style: italic', false)}
      fontFamily={select('font-family', FONTS, FONTS[0])}
    >
      H4 value
    </H4>
    <H5
      color={select('color', COLORS, COLORS[1])}
      bold={boolean('bold', false)}
      italic={boolean('font-style: italic', false)}
      fontFamily={select('font-family', FONTS, FONTS[0])}
    >
      H5 value
    </H5>
    <H6
      color={select('color', COLORS, COLORS[1])}
      bold={boolean('bold', false)}
      italic={boolean('font-style: italic', false)}
      fontFamily={select('font-family', FONTS, FONTS[0])}
    >
      H6 value
    </H6>
    <Text
      color={select('color', COLORS, COLORS[1])}
      bold={boolean('bold', false)}
      italic={boolean('font-style: italic', false)}
      fontFamily={select('font-family', FONTS, FONTS[0])}
      size={select('text font-size', FONT_SIZES, FONT_SIZES[0])}
      crossedOut={boolean('cross-out', false)}
    >
      Text value
    </Text>
  </StyledWrapper>
)

storiesOf('Typography', module)
  .addDecorator(BufferView)
  .addDecorator(ThemeDecorator)
  .add('Typography list', list)

storiesOf('Typography', module)
  .addDecorator(BufferView)
  .addDecorator(ThemeDecorator)
  .add('Typography knobs', knobs)
