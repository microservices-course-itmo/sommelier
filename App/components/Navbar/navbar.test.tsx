import React from 'react'
import { themeRender, fireEvent } from '../../config/test-utils'
import { Navbar, Item } from './navbar'
import { Text } from '../Typography'

describe('Navigation Bar component', () => {
  const onPressMock = jest.fn()
  const iconMock = 'icon.jpg'
  it('Component with only necessary props matches snapshot', () => {
    const component = themeRender(<Navbar>abc</Navbar>)
    expect(component.toJSON()).toMatchSnapshot()
  })
  it('Component with all props matches snapshot', () => {
    const component = themeRender(
      <Navbar>
        <Item onPress={onPressMock} text='abc' icon={iconMock} />
        <Item onPress={onPressMock} text='123' icon={iconMock} />
      </Navbar>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
  it('OnPress function should be called', () => {
    const component = themeRender(
      <Navbar>
        <Item onPress={onPressMock} icon={iconMock} />
      </Navbar>
    )
    const navItem = component.getByTestId('sommelier-nav-item')
    fireEvent.press(navItem)
    expect(onPressMock).toHaveBeenCalled()
  })
  // TODO: Icon and Text testing
  // it('Icon prop assigns correctly', () => {
  //   const component = themeRender(
  //     <Navbar>
  //       <Item onPress={onPressMock} icon={iconMock} />
  //     </Navbar>
  //   )
  //   expect(
  //     component.getByTestId('sommelier-nav-item').children[0].children[0]
  //   ).toBe(iconMock)
  // })
  // it('Text prop assigns correctly', () => {
  //   const component = themeRender(
  //     <Navbar>
  //       <Item onPress={onPressMock} text='abc' icon={iconMock} />
  //     </Navbar>
  //   )
  //   expect(component.getByTestId('sommelier-nav-item').children[1]).toBe(
  //     <Text>abc</Text>
  //   )
  // })
})
