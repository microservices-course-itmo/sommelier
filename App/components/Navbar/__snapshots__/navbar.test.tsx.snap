// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Bar component Component with all props matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "bottom": 0,
        "height": 55.47,
        "justifyContent": "center",
        "left": 0,
        "marginBottom": 0,
        "marginLeft": "auto",
        "marginRight": "auto",
        "marginTop": 0,
        "position": "absolute",
        "right": 0,
      },
    ]
  }
  testID="sommelier-navbar"
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-around",
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID="sommelier-nav-item"
    >
      <View
        style={
          Array [
            Object {
              "height": 20,
              "marginBottom": 6,
              "width": 20,
            },
          ]
        }
      >
        icon.jpg
      </View>
      <Text
        bold={false}
        color="primaryGray"
        italic={false}
        size="sm"
        style={
          Array [
            Object {
              "color": "#666666",
              "fontSize": 10,
              "fontStyle": "normal",
              "fontWeight": "normal",
              "textDecorationColor": "black",
              "textDecorationLine": "none",
              "textDecorationStyle": "solid",
            },
          ]
        }
        testID="sommelier-typography"
      >
        abc
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID="sommelier-nav-item"
    >
      <View
        style={
          Array [
            Object {
              "height": 20,
              "marginBottom": 6,
              "width": 20,
            },
          ]
        }
      >
        icon.jpg
      </View>
      <Text
        bold={false}
        color="primaryGray"
        italic={false}
        size="sm"
        style={
          Array [
            Object {
              "color": "#666666",
              "fontSize": 10,
              "fontStyle": "normal",
              "fontWeight": "normal",
              "textDecorationColor": "black",
              "textDecorationLine": "none",
              "textDecorationStyle": "solid",
            },
          ]
        }
        testID="sommelier-typography"
      >
        123
      </Text>
    </View>
  </View>
</View>
`;

exports[`Navigation Bar component Component with only necessary props matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "bottom": 0,
        "height": 55.47,
        "justifyContent": "center",
        "left": 0,
        "marginBottom": 0,
        "marginLeft": "auto",
        "marginRight": "auto",
        "marginTop": 0,
        "position": "absolute",
        "right": 0,
      },
    ]
  }
  testID="sommelier-navbar"
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-around",
        },
      ]
    }
  >
    abc
  </View>
</View>
`;
